plugins/cost-insights/src/components/ProductInsightsCard/ProductEntityDialog.tsx

```diff
@@ -0,0 +1,188 @@
+        return (
+          <CostGrowthIndicator
+            className={rowStyles}
+            ratio={row.ratio}
+            formatter={amount => formatPercent(Math.abs(amount))}
+          />
+        );
+      default:
+        return <Typography className={rowStyles}>{row.sku}</Typography>;
+    }
+  };
+}
+
+// material-table does not support fixed rows. Override the sorting algorithm
+// to force Total row to bottom by default or when a user sort toggles a column.
+function createSorter(field?: keyof Omit<RowData, 'id'>) {
+  return function rowSort(data1: {}, data2: {}): number {
+    const a = data1 as RowData;
+    const b = data2 as RowData;
+    if (a.id === 'total') return 1;
+    if (b.id === 'total') return 1;
+    if (field === 'sku') return a.sku.localeCompare(b.sku);
+
+    return field
+      ? a[field] - b[field]
+      : b.previous + b.current - (a.previous - a.current);
+  };
+}
+
+const defaultEntity: Entity = {
+  id: null,
+  aggregation: [0, 0],
+  change: { ratio: 0, amount: 0 },
+  entities: [],
+};
+
+type RowData = {
+  id: string;
+  sku: string;
+  previous: number;
+  current: number;
+  ratio: number;
+};
+
+type ProductEntityDialogOptions = Partial<
+  Pick<BarChartOptions, 'previousName' | 'currentName'>
+>;
+
+type ProductEntityDialogProps = {
+  open: boolean;
+  entity?: Entity;
+  options?: ProductEntityDialogOptions;
+  onClose: () => void;
+};
+
+export const ProductEntityDialog = ({
+  open,
+  entity = defaultEntity,
+  options = {},
+  onClose,
+}: ProductEntityDialogProps) => {
+  const classes = useStyles();
+
+  const data = Object.assign(
+    {
+      previousName: 'Previous',
+      currentName: 'Current',
+    },
+    options,
+  );
+
+  const firstColClasses = classnames(classes.column, classes.colFirst);
+  const lastColClasses = classnames(classes.column, classes.colLast);
+
+  const columns: TableColumn[] = [
+    {
+      field: 'sku',
+      title: <Typography className={firstColClasses}>SKU</Typography>,
+      render: createRenderer('sku', classes),
+      customSort: createSorter('sku'),
+    },
+    {
+      field: 'previous',
+      title: (
+        <Typography className={classes.column}>{data.previousName}</Typography>
+      ),
+      align: 'right',
+      render: createRenderer('previous', classes),
+      customSort: createSorter('previous'),
+    },
+    {
+      field: 'current',
+      title: (
+        <Typography className={classes.column}>{data.currentName}</Typography>
+      ),
+      align: 'right',
+      render: createRenderer('current', classes),
+      customSort: createSorter('current'),
+    },
+    {
+      field: 'ratio',
+      title: <Typography className={lastColClasses}>M/M</Typography>,
+      align: 'right',
+      render: createRenderer('ratio', classes),
+      customSort: createSorter('ratio'),
+    },
+  ];
+
+  const rowData: RowData[] = entity.entities
+    .map(e => ({
+      id: e.id || 'Unknown',
+      sku: e.id || 'Unknown',
```

Comment:
Note, "Unknown" is used here in lieu of "Unlabeled". But I doubt this would get rendered at all.

Conceptually a product could have unlabeled costs, and its entities could have unlabeled costs but in practice I don't think that would be often the case. An entity would just be comprised of a bunch of SKUs.